src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -95,39 +95,19 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     }
 
     // Method Description:
-    // - Handler for pressing Enter on TextBox, trigger
-    //   text search
-    // Arguments:
-    // - <none>
-    // Return Value:
-    // - <none>
-    void SearchBoxControl::SetFocusOnTextbox()
+    // - Funnels focus into the textbox and selects it
+    // - Selects all text.
+    void SearchBoxControl::OnGotFocus(/*const IInspectable& sender,*/ Windows::UI::Xaml::RoutedEventArgs const& args)
     {
-        if (TextBox())
+        if (args.OriginalSource() == *this)

```

Comment:
Just to see if I'm understanding this correctly, this condition returns true _only_ if the `SearchBoxControl` directly gained focus (probably from TermControl)? I'm guessing if the user clicked on the `SearchBoxControl::TextBox()` or its many Toggle-able buttons, the OriginalSource would be the `TextBox` or the buttons, so it wouldn't satisfy this condition?

Reply:
Indeed! I wanted to maintain the behavior that we had, where the searchbox didn't re-select all its text when the textbox was re-focused, but did when we entered the control for the first time. :smile: